src/components/interpolatedColorLegend.ts

```diff
@@ -0,0 +1,249 @@
+
+    private _isVertical() {
+      return this._orientation !== "horizontal";
+    }
+
+    public _doRender() {
+      super._doRender();
+
+      var domain = this._scale.domain();
+
+      var textHeight = this._measurer.measure().height;
+      var text0 = this._formatter(domain[0]);
+      var text0Width = this._measurer.measure(text0).width;
+      var text1 = this._formatter(domain[1]);
+      var text1Width = this._measurer.measure(text1).width;
+
+      var ticks = this._scale.ticks();
+      var numSwatches = ticks.length;
+
+      var padding = this._padding;
+
+      var upperLabelShift: Point = { x: 0, y: 0 };
+      var lowerLabelShift: Point = { x: 0, y: 0 };
+      var lowerWriteOptions = {
+                selection: this._lowerLabel,
+                xAlign: "center",
+                yAlign: "center",
+                textRotation: 0
+            };
+      var upperWriteOptions = {
+                selection: this._upperLabel,
+                xAlign: "center",
+                yAlign: "center",
+                textRotation: 0
+            };
+
+      var swatchWidth: number;
+      var swatchHeight: number;
+      var swatchX: (d: any, i: number) => number;
+      var swatchY: (d: any, i: number) => number;
+
+      if (this._isVertical()) {
+        var longestTextWidth = text1Width > text0Width ? text1Width : text0Width;
+        swatchWidth = this.width() - 3 * padding - longestTextWidth;
+        swatchHeight = (this.height() - 2 * padding) / numSwatches;
+        swatchY = (d: any, i: number) => padding + (numSwatches - (i + 1)) * swatchHeight;
+
+        upperWriteOptions.yAlign = "top";
+        upperLabelShift.y = padding;
+        lowerWriteOptions.yAlign = "bottom";
+        lowerLabelShift.y = -padding;
+
+        if (this._orientation === "left") {
+          swatchX = (d: any, i: number) => padding + longestTextWidth + padding;
+          upperWriteOptions.xAlign = "right";
+          upperLabelShift.x = -(padding + swatchWidth + padding);
+          lowerWriteOptions.xAlign = "right";
+          lowerLabelShift.x = -(padding + swatchWidth + padding);
+        } else { // right
+          swatchX = (d: any, i: number) => padding;
+          upperWriteOptions.xAlign = "left";
+          upperLabelShift.x = padding + swatchWidth + padding;
+          lowerWriteOptions.xAlign = "left";
+          lowerLabelShift.x = padding + swatchWidth + padding;
+        }
+      } else { // horizontal
+        swatchWidth = (this.width() - 4 * padding - text0Width - text1Width) / numSwatches;
+        swatchHeight = this.height() - 2 * padding;
+        swatchX = (d: any, i: number) => (padding + text0Width + padding) + i * swatchWidth;
+        swatchY = (d: any, i: number) => padding;
+
+        upperWriteOptions.xAlign = "right";
+        upperLabelShift.x = -padding;
+        lowerWriteOptions.xAlign = "left";
+        lowerLabelShift.x = padding;
+      }
+
+      this._upperLabel.text(""); // clear the upper label
```

Comment:
Would a _writer.clear() or similar accomplish this is in a more highlevel manner?